tensorflow/python/distribute/collective_all_reduce_strategy.py

```diff
@@ -448,6 +448,9 @@ def _make_input_context(self):
     return input_context
 
   def _experimental_distribute_dataset(self, dataset, options):
+    if options and options.replication_mode == distribute_lib.InputReplicationMode.PER_REPLICA:
```

Comment:
Thanks for adding this here. Can you also add it to _experimental_distribute_datasets_from_functions since it is not yet supported in any other strategy than mirrored strategy? (here and in other classes as well)

Reply:
Added to the rest of strategies.